contrib/endpoints/src/api_manager/method_impl.cc

```diff
@@ -51,17 +50,17 @@ void MethodInfoImpl::addAudiencesForIssuer(const string &issuer,
   if (iss.empty()) {
     return;
   }
+
   set<string> &audiences = issuer_audiences_map_[iss];
-  stringstream ss(audiences_list);
-  string audience;
-  // Audience list is comma-delimited.
-  while (getline(ss, audience, ',')) {
-    if (!audience.empty()) {  // Only adds non-empty audience.
-      std::string aud = utils::GetUrlContent(audience);
-      if (!aud.empty()) {
-        audiences.insert(aud);
-      }
+
+  // Audience list is comma-delimited with possible white spaces.
+  char *tokens = strtok(const_cast<char *>(audiences_list.c_str()), " ,");
```

Comment:
Maybe I am missing something here. I thought string are mutable in C++. Also, this is a call be value so I already have a copy of the original string that I can change. You mean you want me to pass by reference and then create another char buffer? I am unable to see the difference unless you think somethin like: str[i] = 'x' is dangerous?